pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -99,44 +114,50 @@ func (a *ipAssigner) loadIPAddresses() error {
 	return nil
 }
 
-// AssignIP ensures the provided IP is assigned to the dummy device.
+// AssignIP ensures the provided IP is assigned to the dummy device and the ARP/NDP responders.
 func (a *ipAssigner) AssignIP(ip string) error {
 	parsedIP := net.ParseIP(ip)
 	if parsedIP == nil {
 		return fmt.Errorf("invalid IP %s", ip)
 	}
 	addr := util.NewIPNet(parsedIP)
 
-	if err := func() error {
-		a.mutex.Lock()
-		defer a.mutex.Unlock()
+	if a.isIPAssigned(ip) {
+		klog.V(2).InfoS("The IP is already assigned", "ip", ip)
+		return nil
+	}
 
-		if a.assignedIPs.Has(ip) {
-			klog.V(2).InfoS("The IP is already assigned", "ip", ip)
-			return nil
+	if parsedIP.To4() != nil && a.arpResponder != nil {
+		if err := a.arpResponder.AddIP(parsedIP); err != nil {
+			return fmt.Errorf("failed to assign IP %v to ARP responder: %v", ip, err)
 		}
+	}
+	if parsedIP.To4() == nil && parsedIP.To16() != nil && a.ndpResponder != nil {
```

Comment:
Maybe use `utilnet.IsIPv6(parsedIP)` to be more readable. There is `utilnet.IsIPv4` too.

Reply:
ok. Thanks.